apps/client/src/pages/home/Home.tsx

```diff
@@ -1,36 +1,34 @@
-import './Home.css';
 import Pokedex from '../../components/Pokedex/Pokedex';
 import Search from '../../components/Search/Search';
-import { useQuery } from 'react-query';
-
-const API_URL = 'https://pokeapi.co/api/v2';
-
-const getPokemons = async () => {
-  const response = await fetch(`${API_URL}/pokemon`);
-  return response.json();
-};
-
-export type PokemonAPISimple = {
-  name: string;
-  url: URL;
-};
+import { usePokemons } from '../../hooks/usePokemons';
+import { useEffect, useState } from 'react';
+import { BasicPokemon } from '../../types/pokemons';
+import { extractUrlPathSegment } from '../../utils/urlExtractor';
+import { URL_ID_SEGMENT_INDEX } from '../../constants/api';
 
 function Home() {
-  const { data, isLoading, isError } = useQuery('pokemons', getPokemons);
+  const { data: pokemonSummaries, isLoading, error } = usePokemons();
+  const [pokemons, setPokemons] = useState<BasicPokemon[]>([]);
+
+  useEffect(() => {
+    if (pokemonSummaries) {
+      const pokemonsList = pokemonSummaries.map((pokemonSummary) => ({
+        name: pokemonSummary.name,
+        // the only way to get id of a Pokémon is to extract it from the URL path. (e.g. /api/v2/pokemon/1)
+        id: extractUrlPathSegment(pokemonSummary.url, URL_ID_SEGMENT_INDEX),
+      }));
+      setPokemons(pokemonsList);
+    }
+  }, [pokemonSummaries]);
 
   if (isLoading) return <h1>Loading...</h1>;
-  if (isError) return <h1>Error...</h1>;
-
-  console.log(data);
-  const pokemonNames: string[] = data.results.map(
-    (r: PokemonAPISimple) => r.name,
-  );
+  if (error || !pokemonSummaries) return <h1>{error}</h1>;
```

Comment:
would be better be like this:
```suggestion
  if (error || isEmptyArray(pokemonSummaries)) return <h1>{error}</h1>;
```